File: Leetcode/twoSum.java
class twoSum {
    public int[] twoSum(int[] nums, int target) {
        for(int i=0; i<nums.length; i++){
            for(int j=i+1; j<nums.length; j++){
                int diff = target - nums[i];
                
                if(nums[j] == diff){
                    return new int[]{i,j};
                }
            }
        }
        throw new IllegalArgumentException("no match found");
    }
}

//1ms solution
class twoSum {
    public int[] twoSum(int[] nums, int target) {
       int[] result = new int[2];
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for(int i=0; i<nums.length; i++){
            int diff = target - nums[i];
            if(map.containsKey(diff)){
                result[0] = i;
                result[1] = map.get(diff);
                return result;
            }
            map.put(nums[i], i);
        }
        return result;
    }
}

File: Leetcode/Golang/twoSum.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)
    ans := make([]int, 2)
    for i:=0; i<len(nums); i++{
        diff := target - nums[i]
        if v, found := m[diff]; found {
            ans[0] = v
            ans[1] = i
            return ans
        }
        m[nums[i]] = i
    }
    return ans
}